dependencies.yaml

```diff
@@ -0,0 +1,101 @@
+# Dependency list for https://github.com/rapidsai/dependency-file-generator
+files:
+  all:
+    output: conda
+    matrix:
+      cuda: ["11.5"]
+      arch: [x86_64]
+    includes:
+      - build
+      - cudatoolkit
+      - develop
+      - doc
+      - run
+      - test_python
+  test_python:
+    output: none
+    includes:
+      - cudatoolkit
+      - py_version
+      - test_python
+  test_cpp:
+    output: none
+    includes:
+      - cudatoolkit
+      - test_cpp 
+  checks:
+    output: none
+    includes:
+      - build
+      - develop
+      - py_version
```

Comment:
Style checks should not be dependent on a specific Python version being specified in the matrix (if no value is specified, the matrix match will fail). The Python requirement in the `build` list should be sufficient. However, we may want a default (empty matrix in the last entry) of `py_version` defining `>=3.8,<3.10` instead of putting it in `build`.
```suggestion
```

Reply:
Any reason doing it just for py_version? Why not also cython?

Reply:
And are you suggesting removing `python` entry from `build` and use `py_version` in `all`?

Reply:
I think you did what I wanted -- remove `py_version` from the style check dependency list.

As for removing `python` from `build` -- is there a C++ library that can be built without Python? If so, I would remove `python`, `cython`, ... from `build` and put them in a separate list like `build_python`. We're trying to achieve a level of modularity/granularity that is useful for minimizing dependencies at each stage of a build.